tests/test_qnode.py

```diff
@@ -1092,6 +1093,34 @@ def circuit(a):
         assert len(res) == 2
         assert dev.shots == 3
 
+    def test_warning_finite_shots_dev(self):

```

Comment:
It would be good to add a more bare-bone execution pipeline test too that doesn't necessarily use QNodes (but uses the execution pipeline the same way) as a QNode does.

Reply:
Added a test that calls `qml.execute` on a tape 👍